ui/app/src/model/project-client.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright 2021 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { fetchJson, Metadata } from '@perses-ui/core';
+import { useQuery, UseQueryOptions } from 'react-query';
+import buildURL from './url-builder';
+
+const resource = 'projects';
+
+export interface ProjectModel {
+  kind: 'Project';
+  metadata: Metadata;
+}
+
+type ProjectListOptions = Omit<
```

Comment:
I'm feeling we will duplicate each time this kind of type. Maybe worth to have one generic.

Reply:
Yeah, good idea. Might be worth making a shorthand generic type for that.

Reply:
I _really_ wish we had some codegen for a bunch of these types based on some API spec. Definitely one thing I miss about not having a GraphQL API + schema. 😂

Reply:
will put that on my todo list :D